### DISCUSSION TITLE
Couchbase support

### DISCUSSION THREAD

**Discussion by ceddybi** [prisma/prisma#24549] [Q&A]
### Question

@janpio @Jolg42 let's say you want to create a custom adapter for a database like Couchbase, does it require to write custom adapters in both rust and typescript?

**Comment by Jolg42** [comment]
@ceddybi I see that https://www.couchbase.com/comparing-couchbase-vs-couchdb/ mentions that CouchDB supports “SQL++ (SQL for JSON)”

I'm not sure what it means (I didn't check more than that).

We only have driver adapters for SQL databases for now, for example https://github.com/prisma/prisma/blob/main/packages/adapter-pg/src/pg.ts

What happens is that the adapter implements some expected functions required by the Prisma Query Engine.
The problem for your case would I think
- Does Couchbase support all the SQL queries that the Prisma Query Engine will generate?
- Is it doable to convert SQL queries to “SQL++ (SQL for JSON)” in the adapter?

**Reply by ceddybi** [reply]
@Jolg42 

> for example https://github.com/prisma/prisma/blob/main/packages/adapter-pg/src/pg.ts

So if i was to create a custom adapter i can copy this and replace with the couchabse client, and change logic / methods, so no need to write rust? only the TS adapter?

> Does Couchbase support all the SQL queries that the Prisma Query Engine will generate?

I think it might support all SQL but it also includes it's SQL++

> Is it doable to convert SQL queries to “SQL++ (SQL for JSON)” in the adapter?

Yes, how would you intercept the generated sql and inject couchbase sql++, is it using the adapter-pg example you just shared, or it'll also require some Rust too?

**Reply by ceddybi** [reply]
@Jolg42 Good day sir

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
